src/components/HistoryTable/hooks/test-utils.tsx

```diff
@@ -0,0 +1,40 @@
+import { useMemo } from "react";
+import { MockedProvider, MockedResponse } from "@apollo/client/testing";
+import { HistoryTableProvider, useHistoryTable } from "../HistoryTableContext";
+
+type UseHistoryTableTestHookType = <T extends (...args: any) => any | any[]>(
+  useHook: T,
+  args: Parameters<T>
+) => {
+  hookResponse: ReturnType<T>;
+  historyTable: ReturnType<typeof useHistoryTable>;
+};
+/** useHistoryTableTestHook takes a hook and useHistoryTable hooks
+ * and combines them into a shared hook which can be rendered under the same wrapper context
+ * and can be used together */
+const useHistoryTableTestHook: UseHistoryTableTestHookType = (
+  useHook,
+  args
+) => {
+  const memoizedArgs = useMemo(() => args, [args]);
+  const hookResponse = useHook(...memoizedArgs);
```

Comment:
would you mind adding more comments around this? I understand after you explained it to me, but it could be helpful to anyone in the future who encounters (and also me when I forget in a month) 😇